src/mercury_engine_data_structures/formats/bctex.py

```diff
@@ -138,20 +148,23 @@
     hash = UInt,
     _mip_map_entries_begin=construct.Seek(construct.this.file_header.mipmap_entries_offset),
     mip_map_entry= construct.Struct(
-        texture_format = construct.Byte,
+        texture_format = construct.Enum(construct.Byte, PICATextureFormat),
         mip_count = construct.Byte,
         compressed = construct.Byte,
         etc1_quality = construct.Byte,
     ),
     image_data = construct.Bytes(construct.this.file_header.texture_data_size)
 )
 
+# this and all related structures is a mixture of some own work + most parts taken from:
+# https://github.com/M-1-RLG/010-Editor-Templates/
+# https://github.com/MapStudioProject/CTR-Studio
+# https://github.com/FanTranslatorsInternational/Kuriimu2/
```

Comment:
Since its not listed, have you looked at htis?
https://www.3dbrew.org/wiki/CTPK

Reply:
No, haven't looked at it. Doesn't seem to have something new?

Reply:
It has some info that's applicable in general but probably never seen in MSR. Like the fact that some entries are aligned with specific sections, meaning they can techncially have variable padding, where some some offsets start or that the texture entry's type is an enum and what they stand for.

Reply:
In general, it might probably be a good idea to have a comment, that in MSR, BCTEX/CTPK only ever have one texture entry?

Reply:
Of course there is variable padding, that's why I use `construct.Seek`. If you want to create a file from scratch with that, you would need much more logic to adjust all the offsets. And yes, many of this "stuff" could be a list of "stuff". These are also the two biggest issue with the ctr studio pr.
I have no ctpk file with more than one entry, don't want to spend the time constructing one and definetly won't do it for MEDS because this part is solely used for MSR. Every CTPK is encapsulated in one BCTEX and BCTEX only has one width, height and name. IMO it's already implicit that BCTEX only contains one image in the CTPK.

Reply:
> that's why I use construct.Seek

Missed tath completely, whoops.

> IMO it's already implicit that BCTEX only contains one image in the CTPK

IDK. If I'd be an outsider trying to research a CTPK format for another game, it'd definitely throw me off. But that's probably not a concern here?